extension/aodn-classifier/src/main/java/au/org/emii/classifier/UriClassifier.java

```diff
@@ -27,10 +28,21 @@ public UriClassifier(ThesaurusFinder thesaurusFinder, String vocabularyScheme, S
         this.indexKey = indexKey;
     }
 
+    public IAodnThesaurus findThesaurus(String scheme) {
+
+        Thesaurus thesaurus = thesaurusFinder.getThesaurusByConceptScheme(scheme);
+        if (thesaurus == null) {
+            return new NullThesaurus();
```

Comment:
Not reporting error here like above.  Probably better to de-duplicate the code (so there's no opportunity for different behaviour) by adding this code to the IF or creating a factory class for it (I'd probably create a factory class for it - I'm not sure about the new fad of adding factory methods to IF's).

Reply:
fixed this now - there was an existing class ThesaurusDirectoryLoader hiding in the tests directory that I added a method to.